src/Shared/Microsoft.DotNet.Kusto/KustoServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DotNet.Kusto
+{
+    public static class KustoServiceCollectionExtensions
+    {
+        public static IServiceCollection AddKustoClientProvider(this IServiceCollection services)
+        {
+            return services.AddSingleton<IKustoClientProvider, KustoClientProvider>();
```

Comment:
I like to make these methods take an Action<OptionsType> so that people don't forget to configure it.  Try and make sure the "AddThing" is a one stop shop that doesn't require a bunch of other "Add" methods to prevent a runtime error.

Reply:
I just followed what we did for the Github.Authentication ServiceCollectionExtensions and Maestro.AzureDevOps. Do you have an example of what you mean here?

Reply:
I'd make the method something like:
```csharp
        public static IServiceCollection AddKustoClientProvider(this IServiceCollection services, Action<ISomethingSomethingProvider, KustoClientOptions> configure)
        {
            services.AddSingleton<IKustoClientProvider, KustoClientProvider>();
            services.Configure(configure);
            return services;
```

That way it's not possible to inject the type and fail to configure it since doing so will lead to a type that doesn't work.  There are 2 lines that are always required for this guy, so let's make sure you always do both.  I like reducing the possibility of misconfiguring DI, since it's so easy to do and we have very little validation around it.

Reply:
Makes sense. Thank you!